### ISSUE TITLE
PartialFailure backup with a CSI plugin

### ISSUE THREAD

**Issue by sbirl** [vmware-tanzu/velero#5180]
**What steps did you take and what happened:**
* `velero backup create testcsibackup2`
  * `velero 1.9` with csi-plugin
  * volumesnapshot-crd
* Created StorageSnapshot class.

**What did you expect to happen:**
Not a partial failure, that's for sure.

**The following information will help us better understand what's going on**:

Please provide the output of the following commands (Pasting long output into a [GitHub gist](https://gist.github.com) or other pastebin is fine.)

* https://gist.github.com/sbirl/0fb9b1a16e812b4a9a702b270047cad9

- `velero backup describe <backupname>` or `kubectl get backup/<backupname> -n velero -o yaml`

```
Name:         testcsibackup2
Namespace:    velero
Labels:       velero.io/storage-location=default
Annotations:  velero.io/source-cluster-k8s-gitversion=v1.23.7
              velero.io/source-cluster-k8s-major-version=1
              velero.io/source-cluster-k8s-minor-version=23

Phase:  PartiallyFailed (run `velero backup logs testcsibackup2` for more information)

Errors:    6
Warnings:  3

Namespaces:
  Included:  *
  Excluded:  <none>

Resources:
  Included:        *
  Excluded:        <none>
  Cluster-scoped:  auto

Label selector:  <none>

Storage Location:  default

Velero-Native Snapshot PVs:  auto

TTL:  720h0m0s

Hooks:  <none>

Backup Format Version:  1.1.0

Started:    2022-08-03 12:52:42 -0400 EDT
Completed:  2022-08-03 13:23:23 -0400 EDT

Expiration:  2022-09-02 12:52:42 -0400 EDT

Total items to be backed up:  4925
Items backed up:              4925

Velero-Native Snapshots: <none included>
```

- `velero backup logs <backupname>`
* see gist

- `velero restore describe <restorename>` or `kubectl get restore/<restorename> -n velero -o yaml`
- `velero restore logs <restorename>`

* No restore was performed.

**Environment:**

- Velero version (use `velero version`): 1.9
- Velero features (use `velero client config get features`): CSI
- Kubernetes version (use `kubectl version`): v1.22 thru v1.24

---

**Comment by Lyndon-Li** [comment]
This is a known issue, some snapshots may take time to get ready after creation, because of the data size of the volume, etc.
Velero needs to wait for the snapshots to get ready, but currently the timeout is 10 minute and cannot be configured. Therefore is already a [PR](https://github.com/vmware-tanzu/velero/pull/5104) to make this timeout configurable.

**Comment by blackpiglet** [comment]
#5048 
A similar issue was created.
I think we can trace this issue in that one.

**Comment by blackpiglet** [comment]
Fix is merged, and it's cherry-picked into v1.9.1 too.
Close it for now. Feel free to open if your issue is not addressed.
